lib/db/mysql.js

```diff
@@ -916,15 +916,27 @@ module.exports = function (log, error) {
 
   // Update : emails
   // Values : uid = $1, email = $2
-  var SET_PRIMARY_EMAIL = 'CALL setPrimaryEmail_3(?, ?)'
+  var SET_PRIMARY_EMAIL = 'CALL setPrimaryEmail_4(?, ?)'
   MySql.prototype.setPrimaryEmail = function (uid, email) {
-    return this.write(
-      SET_PRIMARY_EMAIL,
-      [
-        uid,
-        email
-      ]
-    )
+    return this.getSecondaryEmail(email)
+      .then((emailRecord) => {
+        if (emailRecord.uid.toString('utf8') !== uid.toString('utf8')) {
+          throw error.cannotSetUnownedPrimaryEmail()
+        }
```

Comment:
This is still doing the check outside of a transaction boundary, which means there's a (very small, but non-zero) chance that we could put the db in a bad state.  We could use `this.transaction` here to manage the transaction in code (see the definition of `writeMultiple` for an example).  Or we could find another way to do the check inside the transaction in the stored procedure, perhaps something like:

```
START TRANSACTION;
  SELECT normalizedEmail INTO @foundEmail FROM emails WHERE uid = inUid AND normalizedEmail = inNormalizedEmail AND isVerified;
  IF @foundEmail IS NULL THEN
   SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1062, MESSAGE_TEXT = 'Can not change email. Could not find email.';
  END IF;

  UPDATE emails SET isPrimary = false WHERE uid = inUid AND isPrimary = true;
  UPDATE emails SET isPrimary = true WHERE uid = inUid AND normalizedEmail = inNormalizedEmail;
COMMIT;
```